### PR TITLE
Upload timings to datadog

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Uploads data generated in #261 to datadog

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201707629991362/1201762844341059/f) (_internal_)
- [Datadog API](https://docs.datadoghq.com/api/latest/logs/)

## 🎥 Demo

image

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -313,7 +313,14 @@ jobs:
       - name: Read results
         if: always()
         run: |
-          find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add
+          message=$(find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add)
+          curl -X POST "https://http-intake.logs.datadoghq.com/api/v2/logs" \
+            -H "Content-Type: application/json" \
+            -H "DD-API-KEY: ${{ secres.DATADOG_API_KEY }}" \
```
Comment (nonparibus) [inline .github/workflows/rust.yml:319]: `secres` typo?

Comment (TimDiekmann) [reply .github/workflows/rust.yml:319]: Yes

Comment (TimDiekmann) [inline .github/workflows/rust.yml:319]:
```suggestion
            -H "DD-API-KEY: ${{ secrets.DATADOG_API_KEY }}" \
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -313,7 +313,14 @@ jobs:
       - name: Read results
         if: always()
         run: |
-          find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add
+          message=$(find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add)
+          curl -X POST "https://http-intake.logs.datadoghq.com/api/v2/logs" \
+            -H "Content-Type: application/json" \
+            -H "DD-API-KEY: ${{ secrets.DATADOG_API_KEY }}" \
+            -d $(jq -n \
+                    --argjson message "$message" \
```
Comment (TimDiekmann) [inline .github/workflows/rust.yml:321]:
```suggestion
                    --argjson message "$message" \
```

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -313,7 +313,17 @@ jobs:
...
+          directory="${{ matrix.directory }}"
+          package="${directory##*/}"
+          tags="package:$package"
+          curl -X POST "https://http-intake.logs.datadoghq.com/api/v2/logs" \
+            -H "Content-Type: application/json" \
+            -H "DD-API-KEY: ${{ secrets.DATADOG_API_KEY }}" \
+            -d $(jq -n \
+                    --argjson message "$message" \
+                    --arg ddtags "$tags" \
+                    '{ message: $message, ddsource: "hash-gh-actions", hostname: "github", service: "actions", ddtags: $ddtags }')
```
Comment (Alfred-Mountfield) [inline .github/workflows/rust.yml:326]: Could you push one with this format manually with curl (although maybe change one or two of the keys so they don't appear later on, maybe add test- in front of the source or something)

And then post the example in a screenshot in the PR?

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -313,7 +313,17 @@ jobs:
...
+          directory="${{ matrix.directory }}"
+          package="${directory##*/}"
+          tags="package:$package"
+          curl -X POST "https://http-intake.logs.datadoghq.com/api/v2/logs" \
+            -H "Content-Type: application/json" \
+            -H "DD-API-KEY: ${{ secrets.DATADOG_API_KEY }}" \
+            -d $(jq -cn \
+                    --argjson message "$message" \
+                    --arg ddtags "$tags" \
+                    '{ message: $message, ddsource: "hash-gh-actions", hostname: "github", service: "actions", ddtags: $ddtags }')
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply .github/workflows/rust.yml:333]: I've updated the tags to include meaningful information.

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -310,15 +313,29 @@ jobs:
...
         if: always()
         run: |
-          find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add
+          message=$(find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add)
+          directory="${{ matrix.directory }}"
+          tags="crate:${directory##*/},event:${{ github.event_name }},test:${{ matrix.tests }},stage:dev,toolchain:${{ matrix.toolchain }},user:${{ github.actor }}"
+          if [[ ${{ github.ref_type }} = branch ]]; then
+            tags="$tags,branch:${{ github.ref_name }}"
+          else
+            tags="$tags,tag:${{ github.ref_name }}"
```
Comment (Alfred-Mountfield) [inline .github/workflows/rust.yml:325]: I would perhaps change this just to be a little clearer
```suggestion
            tags="$tags,git_tag:${{ github.ref_name }}"
```

Comment (TimDiekmann) [reply .github/workflows/rust.yml:325]: I'm rethinking this part. Currently, we don't have any releases and we don't plan to add a tag soon-ish. Let's just remove it?
When we do releases, I think, we want a persistent tag `version:*.*.*`?

Comment (Alfred-Mountfield) [reply .github/workflows/rust.yml:325]: sounds good

**Review by Alfred-Mountfield — APPROVED** [review]

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**